autospec/specfiles.py

```diff
@@ -1583,17 +1623,30 @@ def write_ant_pattern(self):
         self._write_strip("")
 
     def write_gradle_pattern(self):
+        """Write gradle build pattern to spec file."""
         self.write_prep()
         self._write_strip("%build")
         self.write_build_prepend()
         self.write_proxy_exports()
-        self._write_strip("mkdir -p /builddir/.m2")
-        self._write_strip("cp -r /usr/share/java/.m2/* /builddir/.m2/")
-        self._write_strip("gradle --offline " + self.extra_make)
+        self._write_strip("mkdir -p %{buildroot}")
+        # It's ok if this doesn't exist
+        self._write_strip("cp -r /usr/share/java/.m2 ~/.m2 || :")
+        if self.subdir:
+            self._write_strip("pushd " + self.subdir)
+
+        # Point to our local maven repo, first
+        # style check does not like the escapes here
+        self._write_strip(r"sed -i 's|\(repositories\s*{\)|\1\n    mavenLocal()|' build.gradle")
+
+        self._write_strip("gradle --offline dependencies")
```

Comment:
Under what conditions is the `dependencies` command supposed to exit with an error? I just tried running `gradle dependencies` in the kafka 2.3.0 source root as a test, and I encounter the error below? Anything I'm missing, or is this expected for now?

```
FAILURE: Build failed with an exception.

* Where:
Build file '/tmp/kafka-2.3.0-src/build.gradle' line: 153

* What went wrong:
A problem occurred evaluating root project 'kafka-2.3.0-src'.
> Failed to apply plugin [id 'org.gradle.signing']
   > Could not create an instance of type org.gradle.plugins.signing.SigningExtension_Decorated.
      > Invalid signature file digest for Manifest main attributes

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.

* Get more help at https://help.gradle.org

BUILD FAILED in 0s
```

Reply:
Hmmm, good question. I only stuck this in there to give us something potentially to scan; it's not needed for the build. We can take it out, or make it opportunistically non-fatal instead.